### PR TITLE
Add split Linux docs jobs to prod builder json

### LINKED ISSUE #78887: linux_docs has not run on prod in recent runs
image

Should this be removed from the prod builders, or be added to the luci config? It's running in try.

### PR DESCRIPTION
**Pull request by jmagman**
The `Linux docs` prod builder was split into [`Linux docs_test`](https://ci.chromium.org/p/flutter/builders/prod/Linux%20docs_test) and [`Linux docs_publish`](https://ci.chromium.org/p/flutter/builders/prod/Linux%20docs_publish) in https://github.com/flutter/infra/pull/375.  Add them to the framework builders json file.

`Linux docs` was showing up as not running:
Screen Shot 2021-03-23 at 11 19 00 AM

Note the try job is still called [`Linux docs`](https://ci.chromium.org/p/flutter/builders/try/Linux%20docs).

Fixes https://github.com/flutter/flutter/issues/78887

### CODE REVIEW

File: dev/try_builders.json [hunk 1]
```diff
@@ -31,20 +31,6 @@
       "task_name": "linux_build_tests_2_2",
       "enabled": true
     },
-    {
```
Comment (jmagman) [inline dev/try_builders.json:34]: Hmm, `Linux docs_publish` and `Linux docs_test` don't seem to be try builders:
https://ci.chromium.org/p/flutter/builders/try/Linux%20docs_test
https://ci.chromium.org/p/flutter/builders/try/Linux%20docs_publish

Just prod:
https://ci.chromium.org/p/flutter/builders/try/Linux%20docs_test
https://ci.chromium.org/p/flutter/builders/try/Linux%20docs_publish

@gspencergoog do you know if this is expected?  Should at least `docs_test` be running on pre-submit?

Comment (jmagman) [reply dev/try_builders.json:34]: Oh!  It's still `Linux docs` in try!  https://ci.chromium.org/p/flutter/builders/try/Linux%20docs

Please disregard the question then, let me revert removing this.

File: dev/prod_builders.json [hunk 2]
```diff
@@ -121,10 +121,16 @@
...
     {
-      "name": "Linux docs",
+      "name": "Linux docs_test",
       "repo": "flutter",
-      "task_name": "linux_docs",
-      "flaky": false
+      "task_name": "linux_docs_test",
+      "flaky": true
+    },
+    {
```
Comment (godofredoc) [inline dev/prod_builders.json:129]: Should Linux docs_publish run only in release branches? adding it here will run it for master.

Comment (jmagman) [reply dev/prod_builders.json:129]: I think it's supposed to run on master, just not the release candidate branches:
https://github.com/flutter/infra/blob/e83d25832f6b56a7651365076eb842f57328343a/config/packaging_config.star#L149-L152
```
    # Docs builds are only done on master and stable channels
    if branch in ("master", "stable"):
        common.linux_prod_builder(
            name = "Linux%s docs_publish|docs" % ("" if branch == "master" else " " + branch),
```
I'll wait until @christopherfujino comes back from vacation, he can validate that's what he intended.

Comment (christopherfujino) [reply dev/prod_builders.json:129]: this change is correct, provided we follow-up to mark un-flaky.

confusingly, docs are published only on master (https://master-api.flutter.dev) and stable channels (https://api.flutter.dev).

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**